src/rp2_common/hardware_spi/include/hardware/spi.h

```diff
@@ -197,27 +207,34 @@ static inline void spi_set_slave(spi_inst_t *spi, bool slave) {
  *  \ingroup hardware_spi
  *
  * \param spi SPI instance specifier, either \ref spi0 or \ref spi1
- * \return 0 if no space is available to write. Non-zero if a write is possible
+ * \return false if no space is available to write. True if a write is possible
  *
- * \note Although the controllers each have a 8 deep TX FIFO, the current HW implementation can only return 0 or 1
- * rather than the space available.
  */
-static inline size_t spi_is_writable(spi_inst_t *spi) {
-    // PL022 doesn't expose levels directly, so return values are only 0 or 1
-    return (spi_get_hw(spi)->sr & SPI_SSPSR_TNF_BITS) >> SPI_SSPSR_TNF_LSB;
+static inline bool spi_is_writable(const spi_inst_t *spi) {
+    return (spi_get_hw(spi)->sr & SPI_SSPSR_TNF_BITS);
 }
 
 /*! \brief Check whether a read can be done on SPI device
  *  \ingroup hardware_spi
  *
  * \param spi SPI instance specifier, either \ref spi0 or \ref spi1
- * \return Non-zero if a read is possible i.e. data is present
+ * \return true if a read is possible i.e. data is present
  *
  * \note Although the controllers each have a 8 deep RX FIFO, the current HW implementation can only return 0 or 1
```

Comment:
Should this "note" be removed too, to match the change you just did for `spi_is_writable`? :slightly_smiling_face:

Reply:
xD very fussy, but you're right!